### PR TITLE
Refs #34118, Refs #34504 -- Added backward incompatibility note about EmailBackend.ssl_context.

### PR DESCRIPTION
**Pull request by felixxm**
Follow up to 2848e5d0ce5cf3c31fe87525536093b21d570f69.

ticket-34504

### CODE REVIEW

**Review by hramezani — APPROVED** [review]
👍

File: docs/releases/4.2.txt [hunk 1]
```diff
@@ -467,6 +467,12 @@ Miscellaneous
 
 * Support for ``PROJ`` < 5 is removed.
 
+* :class:`~django.core.mail.backends.smtp.EmailBackend` now verifies a
+  :py:attr:`hostname <ssl.SSLContext.check_hostname>` and
+  :py:attr:`certificates <ssl.SSLContext.verify_mode>`. If you need the
+  previous behavior that is less restrict and not recommended, subclass
```
Comment (sarahboyce) [inline docs/releases/4.2.txt:473]:
```suggestion
  previous behavior that is less restrictive and not recommended, subclass
```

**Review by sarahboyce — APPROVED** [review]

File: docs/releases/4.2.txt (hunk 1 above)
Comment (felixxm) [reply docs/releases/4.2.txt]: Updated :+1:

**Comment by felixxm** [comment]
Thanks both for checking :+1:

**Pull request merged**